cpu-profiling/cl-position-handler/README.md

```diff
@@ -0,0 +1,357 @@
+A single instance of position handler processed around **163 ops/sec** and the average duration for each message is **6.38ms**
+
+### Commit message handling
+
+Dumped 50k commit kafka messages on position topic and captured the grafana dashboards `dashboard-images/baseline-comm,it`.
+A single instance of position handler processed around **394 ops/sec** and the max duration for processing message is **8.03ms** and it reduced over the time and minimum duration is **2.47ms**
+
+---
+
+## Profile using chrome debugger
+- Run **position handler** using `node --inspect=0.0.0.0:9229 src/handlers/index.js handler --position`
+- Use **Chrome Debugger** to connect to this instance and start CPU profiling
+- Trigger prepare / commit messages
+- Stop profiling and observe the time taken for various operations in the code
+
+### Observations
+#### Profile: CPU-20230817T142002.cpuprofile
+- The following lines of code seems taking more time in `models/position/facade.js`
+  ```
+  reservedTransfers[transfer.transferId] = { transferState, transfer, rawMessage, transferAmount }
+
+  processedTransfers[transferId] = { transferState, transfer, rawMessage, transferAmount, runningPosition, runningReservedValue }
+  ```
+  Initially, I thought if the code is copying any buffer content instead of referencing it but couldn't find any buffer content in the message afterwards.
+
+## Profiling using Clinic.js bubbleprof
+- Install **Clinic.js** tool
+- Stop the central ledger position handler node service
+- Start the service using Clinic.js
+  ```
+  clinic bubbleprof -- node src/handlers/index.js handler --position
+  ```
+- Run the **k6** load
+- Terminate the position handler and it opens up an html file immediately.
+- Sometimes you may see an error `Analysing dataError: premature close`, no need to worry in this case. The profile data is already saved in the current folder `.clinic`. We can generate the html report from the profile dump using the following command.
+  ```
+  clinic bubbleprof --visualize-only .clinic/<PID>.clinic-bubbleprof
+  ```
+
+### Observations
+#### Profile: 892321.clinic-bubbleprof.html
+- Observed `23,659 ms in asynchronous delays, 19,145 ms in synchronous delays.` in mysql connect calls
+  ```
+  at Socket.Socket.connect node:net:993:7
+  at Connection.connect ../../node_modules/mysql/lib/Connection.js:76:13
+  at  ../../node_modules/@mojaloop/central-services-database/node_modules/knex/lib/dialects/mysql/index.js:56:18
+  at Client_MySQL.acquireRawConnection ../../node_modules/@mojaloop/central-services-database/node_modules/knex/lib/dialects/mysql/index.js:51:12
+  at create ../../node_modules/@mojaloop/central-services-database/node_modules/knex/lib/client.js:237:39
+  ```
+  By looking at the query log in mysql, the connections are being reused, so it seems like there is no issue.
+
+---
+
+## Mysql queries
+
+The following are the queries executed during a prepare / commit message handling operation.
+
+### Prepare
```

Comment:
Can you also indicate what queries are cached when caching is enabled for both prepare and fulfil (i.e. commit) please.